### PR TITLE
Javadoc to describe the block/fluid state properties Minecraft provides

### PR DESCRIPTION
**Pull request by i509VCB**

### CODE REVIEW

**Review by liach — CHANGES_REQUESTED** [review]
Please move all `@param` to comments on individual parameters than as part of the method comment.

**Comment by i509VCB** [comment]
> 
> 
> Please move all `@param` to comments on individual parameters than as part of the method comment.

Okay so thats why cloak didn't complete them properly

**Comment by i509VCB** [comment]
Fixed the param comments

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 1]
```diff
@@ -1,94 +1,204 @@
...
 	FIELD field_12504 WEST_WIRE_CONNECTION Lnet/minecraft/class_2754;
+		COMMENT A property that specifies how redstone wire attaches to the west.‌
 	FIELD field_12505 BITES Lnet/minecraft/class_2758;
+		COMMENT A property that specifies the bites taken out of a cake.
 	FIELD field_12506 CHEST_TYPE Lnet/minecraft/class_2754;
+		COMMENT A property that specifies what type of chest a block is.
 	FIELD field_12507 RAIL_SHAPE Lnet/minecraft/class_2754;
+		COMMENT A property that specifies the two directions a rail connects to.
 	FIELD field_12508 WATERLOGGED Lnet/minecraft/class_2746;
+		COMMENT  A property that specifies if a block is waterlogged.
```
Comment (liach) [inline mappings/net/minecraft/state/property/Properties.mapping]:
```suggestion
		COMMENT A property that specifies if a block is waterlogged.
```

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 2]
```diff
@@ -1,94 +1,204 @@
 CLASS net/minecraft/class_2741 net/minecraft/state/property/Properties
+	COMMENT Contains all block and fluid state properties that Minecraft uses.
 	FIELD field_12480 FALLING Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a fluid is falling.
 	FIELD field_12481 HORIZONTAL_FACING Lnet/minecraft/class_2753;
+		COMMENT A property that specifies the direction a block is facing.
+		COMMENT
+		COMMENT <p>This property only allows a block to face in one of the cardinal directions (north, south, east and west).
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Properties.mapping:8]: Closing tag

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 3]
```diff
@@ -1,94 +1,204 @@
...
 	FIELD field_12488 EYE Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if an end portal frame contains an eye of ender.
 	FIELD field_12489 NORTH Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if this block is connected to another block from the north.
 	FIELD field_12490 LEVEL_1_8 Lnet/minecraft/class_2758;
+		COMMENT A property that specifies the height of a fluid on a scale of 1 to 8.
 	FIELD field_12491 IN_WALL Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a fence gate is attached to a wall.
+		COMMENT
+		COMMENT <p>This lowers the fence gate by 3 pixels to attach more cleanly to a wall.
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Properties.mapping:30]: Closing tag

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 4]
```diff
@@ -1,94 +1,204 @@
...
 	FIELD field_12492 PISTON_TYPE Lnet/minecraft/class_2754;
+		COMMENT A property that specifies what type of piston.
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Properties.mapping]:
```suggestion
		COMMENT A property that specifies the type of a piston.
```

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 5]
```diff
@@ -1,94 +1,204 @@
...
 	FIELD field_12493 ATTACHED Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a tripwire is attached to tripwire hook.
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Properties.mapping]:
```suggestion
		COMMENT A property that specifies if a tripwire is attached to a tripwire hook.
```

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 6]
```diff
@@ -1,94 +1,204 @@
...
 	FIELD field_12497 AGE_3 Lnet/minecraft/class_2758;
+		COMMENT A property that specifies the age of a block on a scale of 0 to 3.
 	FIELD field_12498 AGE_15 Lnet/minecraft/class_2758;
+		COMMENT A property that specifies the age of a block on a scale of 0 to 15.
 	FIELD field_12499 INSTRUMENT Lnet/minecraft/class_2754;
+		COMMENT A property that specifies what instrument a note block will play.
 	FIELD field_12500 HAS_BOTTLE_1 Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a brewing stand has a bottle in slot 1.
 	FIELD field_12501 INVERTED Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a daylight sensor's output is inverted,.
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Properties.mapping]:
```suggestion
		COMMENT A property that specifies if a daylight sensor's output is inverted.
```

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 7]
```diff
@@ -1,94 +1,204 @@
...
 	FIELD field_12505 BITES Lnet/minecraft/class_2758;
+		COMMENT A property that specifies the bites taken out of a cake.
 	FIELD field_12506 CHEST_TYPE Lnet/minecraft/class_2754;
+		COMMENT A property that specifies what type of chest a block is.
 	FIELD field_12507 RAIL_SHAPE Lnet/minecraft/class_2754;
+		COMMENT A property that specifies the two directions a rail connects to.
 	FIELD field_12508 WATERLOGGED Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a block is waterlogged.
 	FIELD field_12509 EGGS Lnet/minecraft/class_2758;
+		COMMENT A property that specifies the amount eggs in a turtle egg block.
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Properties.mapping]:
```suggestion
		COMMENT A property that specifies the amount of eggs in a turtle egg block.
```

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 8]
```diff
@@ -1,94 +1,204 @@
...
 	FIELD field_12511 POWER Lnet/minecraft/class_2758;
+		COMMENT A property that specifies the redstone power of a block.
 	FIELD field_12512 SNOWY Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a block is covered in snow.
 	FIELD field_12513 LEVEL_3 Lnet/minecraft/class_2758;
+		COMMENT A property that specifies how many levels of water there are in a cauldron.
 	FIELD field_12514 PERSISTENT Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a block is persistent.
+		COMMENT
+		COMMENT <p>In vanilla, this is used to specify whether leaves should disappear when the logs are removed.
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Properties.mapping:78]: Closing tag

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 9]
```diff
@@ -1,94 +1,204 @@
...
 	FIELD field_12526 DRAG Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a bubble column should drag entities downwards.
 	FIELD field_12527 WEST Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if this block is connected to another block from the west.
 	FIELD field_12528 OCCUPIED Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a bed is occupied.
 	FIELD field_12529 HORIZONTAL_AXIS Lnet/minecraft/class_2754;
+		COMMENT  A property that specifies the axis a block is oriented to.
+		COMMENT
+		COMMENT <p>This property only allows a block to be oriented to the X and Z axes.
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Properties.mapping:110]: Closing tag

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 10]
```diff
@@ -1,94 +1,204 @@
...
 	FIELD field_12530 HATCH Lnet/minecraft/class_2758;
+		COMMENT A property that specifies how close an egg is hatching.
 	FIELD field_12531 HAS_BOTTLE_2 Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a brewing stand has a bottle in slot 2.
 	FIELD field_12532 ROTATION Lnet/minecraft/class_2758;
+		COMMENT A property that specifies the rotation of a block on a 0 to 15 scale.
+		COMMENT
+		COMMENT <p>Each rotation is 22.5 degrees.
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Properties.mapping:118]: Closing tag

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 11]
```diff
@@ -1,94 +1,204 @@
...
 	FIELD field_12533 DOUBLE_BLOCK_HALF Lnet/minecraft/class_2754;
+		COMMENT A property that specifies whether a double height block is the upper or lower half..
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Properties.mapping]:
```suggestion
		COMMENT A property that specifies whether a double height block is the upper or lower half.
```

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 12]
```diff
@@ -1,94 +1,204 @@
...
 	FIELD field_12534 COMPARATOR_MODE Lnet/minecraft/class_2754;
+		COMMENT A property that specifies the mode a comparator is set to.
 	FIELD field_12535 SHORT Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a piston head is shorter than normal.
 	FIELD field_12536 LAYERS Lnet/minecraft/class_2758;
+		COMMENT A property that specifies how many layers of snow are in a snow block.
 	FIELD field_12537 OPEN Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a block is open.
+		COMMENT
+		COMMENT <p>This property is normally used for doors, trapdoors and fence gates but is also used by barrels.
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Properties.mapping:130]: Closing tag

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 13]
```diff
@@ -1,94 +1,204 @@
...
 	FIELD field_12538 LEVEL_15 Lnet/minecraft/class_2758;
 	FIELD field_12539 UNSTABLE Lnet/minecraft/class_2746;
+		COMMENT  A property that specifies if TNT is unstable.
+		COMMENT
+		COMMENT  <p>In vanilla, if TNT is unstable, it will ignite when broken.
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Properties.mapping:135]: Closing tag

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 14]
```diff
@@ -1,94 +1,204 @@
...
 	FIELD field_12540 SOUTH Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if this block is connected to another block from the south.
 	FIELD field_12541 DISTANCE_1_7 Lnet/minecraft/class_2758;
+		COMMENT A property that specifies the overhang distance of a block on a scale of 1-7.
 	FIELD field_12542 STRAIGHT_RAIL_SHAPE Lnet/minecraft/class_2754;
+		COMMENT A property that specifies the two directions a rail connects to.
+		COMMENT
+		COMMENT <p>This property does not allow for a rail to turn.
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Properties.mapping:143]: Closing tag

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 15]
```diff
@@ -1,94 +1,204 @@
...
 	FIELD field_12543 PICKLES Lnet/minecraft/class_2758;
+		COMMENT A property that specifies how many pickles are in a sea pickle.
 	FIELD field_12544 HAS_RECORD Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a jukebox has a record.
 	FIELD field_12545 HOPPER_FACING Lnet/minecraft/class_2753;
+		COMMENT A property that specifies the direction a hopper's output faces.
+		COMMENT
+		COMMENT <p>This property does not allow the hopper's output to face upwards.
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Properties.mapping:151]: Closing tag

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 16]
```diff
@@ -1,94 +1,204 @@
...
 	FIELD field_12553 DISARMED Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a tripwire has been disarmed.
 	FIELD field_12554 HAS_BOTTLE_0 Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a brewing stand has a bottle in slot 0.
 	FIELD field_12555 WALL_MOUNT_LOCATION Lnet/minecraft/class_2754;
+		COMMENT A property that specifies the type of wall a block is attached to.
 	FIELD field_12556 AGE_2 Lnet/minecraft/class_2758;
+		COMMENT A property that specifies the age of a block on a scale of 0 to 2.
 	FIELD field_16503 DISTANCE_0_7 Lnet/minecraft/class_2758;
+		COMMENT A property that specifies the overhang distance of a cauldron.
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Properties.mapping]:
```suggestion
		COMMENT A property that specifies the overhang distance of a scaffolding.
```

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 17]
```diff
@@ -1,94 +1,204 @@
...
 	FIELD field_16562 BOTTOM Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a scaffolding block is bottom of a floating segment.
 	FIELD field_17104 ATTACHMENT Lnet/minecraft/class_2754;
+		COMMENT A property that specifies how a bell is attached to a block.
 	FIELD field_17393 HAS_BOOK Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a lectern has a book.
 	FIELD field_17394 SIGNAL_FIRE Lnet/minecraft/class_2746;
+		COMMENT A property that specifies if a campfire's smoke should be taller.
+		COMMENT
+		COMMENT <p>This occurs when a hay bale is placed under the campfire.
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Properties.mapping:187]: Closing tag

File: mappings/net/minecraft/state/property/Properties.mapping [hunk 18]
```diff
@@ -1,94 +1,204 @@
...
 	FIELD field_17586 LEVEL_8 Lnet/minecraft/class_2758;
+		COMMENT A property that specifies how the level of a composter.
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Properties.mapping]:
```suggestion
		COMMENT A property that specifies the level of a composter.
```

File: mappings/net/minecraft/state/property/Properties.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/state/property/Properties.mapping:8]: Nah, javadoc generator will auto add closing tags

File: mappings/net/minecraft/state/property/Properties.mapping (hunk 3 above)
Comment (liach) [reply mappings/net/minecraft/state/property/Properties.mapping:30]: No need

File: mappings/net/minecraft/state/property/Properties.mapping (hunk 8 above)
Comment (liach) [reply mappings/net/minecraft/state/property/Properties.mapping:78]: No need

**Review by YanisBft — APPROVED** [review]

**Pull request merged**